src/scheduler/server_info.cpp

```diff
@@ -980,9 +976,15 @@ free_server_psets(std::unordered_map<std::string, node_partition *>& spsets)
 void
 server_info::dup_server_psets(const std::unordered_map<std::string, node_partition*>& spsets)
 {
-
-	for (const auto& spset : spsets)
-		svr_to_psets[spset.first] = dup_node_partition(spset.second, this);
+	for (const auto& spset : spsets) {
+		node_partition *temp = dup_node_partition(spset.second, this);
+		if (temp != NULL)
+			svr_to_psets[spset.first] = temp;
```

Comment:
You can shrink this code a little by getting rid of temp and assigning directly into svr_to_psets.  You can then check svr_to_psets[spset.first] == NULL